src/type-inference.md

```diff
@@ -71,8 +71,8 @@ inference works, or perhaps this blog post on
 
 [Unification in the Chalk project]: http://smallcultfollowing.com/babysteps/blog/2017/03/25/unification-in-chalk-part-1/
 
-All told, the inference context stores four kinds of inference variables (as of
-<!-- date: 2018-01 --> January 2018):
+All told, the inference context stores four kinds of inference variables
+(as of <!-- date: 2018-01 --> January 2018):
```

Comment:
It turns out that mdBook freaks out when a line begins with an HTML comment:

![image](https://user-images.githubusercontent.com/37223377/108614915-ae90d680-73b3-11eb-86bc-cf3ef1a0fc62.png)

Not sure if it's a bug or expected behavior.

Reply:
It generates this HTML:

```html
<p>All told, the inference context stores four kinds of inference variables (as of</p>
<!-- date: 2018-01 --> January 2018):
```

Probably pulldown-cmark is the one doing that.